lib/Edited/prototype/listen.js

```diff
@@ -1,18 +1,43 @@
 'use strict'
-var on
-var TEXT_ALTERING_EVENT_TYPES
-
-on = require('add-event-handler')
-TEXT_ALTERING_EVENT_TYPES = require('../../text-altering-event-types')
+var on = require('add-event-handler')
+var Combokeys = require('combokeys')
+var keycode = require('keycode')
 
 module.exports = function () {
   var self = this
 
-  self.listener = function (e) {
-    self.discern(e)
+  // a combokeys instance for character addition
+  self.anyCharCombokeys = new Combokeys(self.element)
+  // `any-character` is a combokeys special argument that
+  // calls back on any character
+  self.anyCharCombokeys.bind('any-character', function () {
+    self.discern('character-addition')
+  })
+  self.anyCharCombokeys.stopCallback = function (e) {
+    if (e.charCode === keycode('space')) {
+      // space is handled as a different edit type
```

Comment:
@winhamwr instead of writing our own constants, the job is outsourced to the popular [keycode](https://www.npmjs.com/package/keycode).